src/doom/m_menu.c

```diff
@@ -873,8 +873,8 @@ static void M_DrawSaveLoadBottomLine(void)
     struct stat st;
     char filedate[32];
 
-    stat(P_SaveGameFile(itemOn), &st);
-
+    if (M_stat(P_SaveGameFile(itemOn), &st) != -1)
```

Comment:
`stat` returns `0` on success, so I think it is more logical to write `if (M_stat(P_SaveGameFile(itemOn), &st) == 0)`

Reply:
I see now, thanks, a return value `0` [means success](https://pubs.opengroup.org/onlinepubs/007904975/functions/stat.html). Sure, it looks definitely better.

Made a brief test on Linux, save game path for Strife seems to be fine:
![image](https://github.com/user-attachments/assets/f81ff219-93f6-4fad-9795-2fa5c65ab27f)

Reply:
On Linux, standard C functions support Unicode, it's just a Windows problem. On Windows, saves are stored in the executable directory without a path, so it will work in most cases anyway, unless the user changes the save directory path.

Reply:
Worth to think more then, as current approach turns out not to be optimal. Probably worth to use `savedir` variable to provide a path, and even better to check it for all four games. I'll have to investigate and experement.

Reply:
By trying to create a new game, from desktop folder with Cyrillic symbols and custom savegame folder with Cyrillic symbols as well: `C:\Users\Pusk\Desktop\криспи\crispy-strife.exe -savedir "C:\Users\Pusk\Desktop\сохранения страйф"`

Pretty crazy scenario, but looks likes something have to be updated on Chocolate Doom strife, otherwise:
![image](https://github.com/user-attachments/assets/d7bd10a3-52c6-492c-9314-3ed1eaada6d4)

Using a Latin-only symbols like `C:\Users\Pusk\Desktop\crispy\crispy-strife.exe -savedir "C:\Users\Pusk\Desktop\strife saves"` working perfectly fine and savegame date is appearing, for all four games.

So, `savegamedir` is working fine in `P_SaveGameFile` [here](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/doom/p_saveg.c#L75). It's a bit more complicated in Raven code's `SlotStatus`, but in the end, it also refers to to `savegamedir` in `SV_Filename` [here](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/heretic/p_saveg.c#L51).

Guess, it's all fine now in terms of this PR. Sorry for inconvenience, it's a second time when Strife dead functions playing bad things on me, first one was a memory leak situation in crossfade effect.

Reply:
@JNechaevsky Is `-savedir` with non-Latin paths broken in all four games, or just Strife?

Reply:
It's just Strife. It looks like a bug in the implementation of `win_opendir.c` (also `dirent.h` in MSYS, since it is based on it). If `I_StartGlob` is called with `*` glob and a non-Latin path, `.` and `..` are returned as filenames. Simple patch:
```diff
diff --git a/src/strife/m_saves.c b/src/strife/m_saves.c
index 561ec6a2..797b0e12 100644
--- a/src/strife/m_saves.c
+++ b/src/strife/m_saves.c
@@ -123,7 +123,7 @@ void FromCurr(void)
     {
         byte *filebuffer;
         int filelen;
-        const char *srcfilename;
+        const char *srcfilename, *basename;
         char *dstfilename;
 
         srcfilename = I_NextGlob(glob);
@@ -131,8 +131,13 @@ void FromCurr(void)
         {
             break;
         }
+        basename = M_BaseName(srcfilename);
+        if (basename[0] == '.')
+        {
+            continue;
+        }
 
-        dstfilename = M_SafeFilePath(savepath, M_BaseName(srcfilename));
+        dstfilename = M_SafeFilePath(savepath, basename);
 
         filelen = M_ReadFile(srcfilename, &filebuffer);
         M_WriteFile(dstfilename, filebuffer, filelen);
@@ -165,7 +170,7 @@ void ToCurr(void)
     {
         byte *filebuffer;
         int filelen;
-        const char *srcfilename;
+        const char *srcfilename, *basename;
         char *dstfilename;
 
         srcfilename = I_NextGlob(glob);
@@ -173,8 +178,13 @@ void ToCurr(void)
         {
             break;
         }
+        basename = M_BaseName(srcfilename);
+        if (basename[0] == '.')
+        {
+            continue;
+        }
 
-        dstfilename = M_SafeFilePath(savepathtemp, M_BaseName(srcfilename));
+        dstfilename = M_SafeFilePath(savepathtemp, basename);
 
         filelen = M_ReadFile(srcfilename, &filebuffer);
         M_WriteFile(dstfilename, filebuffer, filelen);
```
We can rewrite `win_opendir.c` using the Unicode Win API.